tasks/utils.js

```diff
@@ -0,0 +1,54 @@
+module.exports = function(grunt) {
+
+    var Q = require("q");
+    // add a few grunt-specific convenience functions to Promise
+    (function() {
+        var thenLogWithFunc = function (func) {
+            return function () {
+                var logArgs = arguments;
+                return this.then(function () {
+                    func.apply(undefined, logArgs);
+                    return arguments.length > 0 ? arguments[0] : undefined;
+                });
+            };
+        };
+        Q.makePromise.prototype.thenLog = thenLogWithFunc(grunt.log.writeln);
+        Q.makePromise.prototype.thenLogOk = thenLogWithFunc(function(){grunt.log.ok("OK")});
+        Q.makePromise.prototype.thenLogVerbose = thenLogWithFunc(grunt.verbose.writeln);
+    })();
+
+    // add support for promises to simpleGit
+    // use it by calling git.promising('checkout', '0.1.0')
+    var git = require('simple-git/src/git');
+    git.prototype.promising = function(fnName) {
+        var logMessage = ("git " + fnName + " " + Array.prototype.slice.call(arguments, 1).join()).trim() + "...";
+        grunt.log.write(logMessage);
+        var deferred = Q.defer();
+        var args = Array.prototype.slice.call(arguments, 1);
+        args.push(function(err, data) {
```

Comment:
Since you go full on with the Q dependency, this function can probably be simplified by just using (logging to be added):

```
var promisified = Q.denodeify(this[fnName].bind(this));
return promisified(Array.prototype.slice.call(arguments, 1));
```